src/main/java/com/razorpay/ApiClient.java

```diff
@@ -113,6 +129,14 @@ public void delete(String path, JSONObject requestObject) throws RazorpayExcepti
     }
 
     if (statusCode >= STATUS_OK && statusCode < STATUS_MULTIPLE_CHOICE) {
+    	
+      if(!responseJson.has(ENTITY)) {
+    	   String cls = StaticEntity(response);
```

Comment:
StaticEntity method should be populateEntiryFromURL(response.request().url().pathSegments())